packages/toolkit/src/view/model/CreateModelForm1.tsx

```diff
@@ -0,0 +1,53 @@
+import { z } from "zod";
+import { InstillErrors } from "../../constant";
+import { validateInstillID } from "../../server";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Nullable, useModelDefinitions } from "../../lib";
+
+// "model_definition": "model-definitions/container",
+
+export type CreateModelFormProps = {
+  accessToken: Nullable<string>;
+  enabledQuery: boolean;
+}
+
+const CreateModelSchema = z
+  .object({
+    id: z.string(),
+    description: z.string().optional().nullable(),
+    visibility: z.enum(["VISIBILITY_PRIVATE", "VISIBILITY_PUBLIC"]).default("VISIBILITY_PRIVATE"),
+    region: z.enum(["REGION_GCP_EUROPE_WEST_4"]).default("REGION_GCP_EUROPE_WEST_4"),
+    hardware: z.enum(["CPU"]).default("CPU"),
+    task: z.enum(["TASK_CLASSIFICATION", "TASK_DETECTION", "TASK_KEYPOINT", "TASK_OCR", "TASK_INSTANCE_SEGMENTATION", "TASK_SEMANTIC_SEGMENTATION", "TASK_TEXT_GENERATION", "TASK_TEXT_TO_IMAGE", "TASK_IMAGE_TO_IMAGE", "TASK_IMAGE_TO_TEXT"]).default("TASK_CLASSIFICATION"),
```

Comment:
@orangecoloured Could you help me extract these enum into the constant folder, in this way we can make it more maintainable

Reply:
Let's resolve this in the next PR. Sorry I spilled some create form work into this PR.